src/distributed_planner/distributed_config.rs

```diff
@@ -1,61 +1,145 @@
-use datafusion::common::extensions_options;
-use datafusion::config::ConfigExtension;
-use datafusion::prelude::SessionConfig;
-use std::fmt::Debug;
+use crate::channel_resolver_ext::ChannelResolverExtension;
+use crate::distributed_planner::task_estimator::CombinedTaskEstimator;
+use crate::{BoxCloneSyncChannel, ChannelResolver, TaskEstimator};
+use arrow_flight::flight_service_client::FlightServiceClient;
+use async_trait::async_trait;
+use datafusion::common::utils::get_available_parallelism;
+use datafusion::common::{DataFusionError, extensions_options, not_impl_err, plan_err};
+use datafusion::config::{ConfigExtension, ConfigField, ConfigOptions, Visit};
+use std::fmt::{Debug, Formatter};
+use std::sync::Arc;
+use url::Url;
 
 extensions_options! {
+    /// Configuration for the distributed planner.
     pub struct DistributedConfig {
-        /// Upon shuffling data, this defines how many tasks are employed into performing the shuffling.
-        /// ```text
-        ///  ( task 1 )  ( task 2 ) ( task 3 )
-        ///      ▲           ▲          ▲
-        ///      └────┬──────┴─────┬────┘
-        ///       ( task 1 )  ( task 2 )       N tasks
-        /// ```
-        /// This parameter defines N
-        pub network_shuffle_tasks: Option<usize>, default = None
-        /// Upon merging multiple tasks into one, this defines how many tasks are merged.
-        /// ```text
-        ///              ( task 1 )
-        ///                  ▲
-        ///      ┌───────────┴──────────┐
-        ///  ( task 1 )  ( task 2 ) ( task 3 )  N tasks
-        /// ```
-        /// This parameter defines N
-        pub network_coalesce_tasks: Option<usize>, default = None
+        /// Sets the maximum amount of files that will be assigned to each task. Reducing this
+        /// number will spawn more tasks for the same number of files.
+        pub files_per_task: usize, default = files_per_task_default()
+        /// Task multiplying factor for when a node declares that it changes the cardinality
+        /// of the data:
+        /// - If a node is increasing the cardinality of the data, this factor will increase.
+        /// - If a node reduces the cardinality of the data, this factor will decrease.
+        /// - In any other situation, this factor is left intact.
+        pub cardinality_task_count_factor: f64, default = cardinality_task_count_factor_default()
```

Comment:
Can this stuff be set via session properties in datafusion sql? Asking so I can fuzz this.